spec/models/miq_expression_spec.rb

```diff
@@ -1631,4 +1631,146 @@
       end
     end
   end
+
+  describe "#sql_supports_atom?" do
+    context "expression key is 'CONTAINS'" do
+      context "operations with 'tag'" do
+        it "returns true if format of field is /\w+\.\w+/ - interpreted as tag of the main model" do
```

Comment:
1. I wanted to emphasize on places where no model analysis involved, which should be hints for future refactoring. Specs (from my point of view) are the same as code -  should be modified  and refactored of code refactored.  
In the same test - does it make sense to pass correct expression to MiqExpression constructor as it is done currently in code or nil to highlight that  instance variable is irrelevant for this case ?
The same with parameter for method - should it be legitimate expression (as in comment) or Hash where only key "tag" is taken in consideration by implementation?

2. Terminology is challenging :). Both (TAG and FIELD) works with fields. 

@gtanzillo Gregg, what do you think of above ?